internal/compiler-interface/src/main/java/xsbti/compile/DefinesClass.java

```diff
@@ -1,12 +1,14 @@
 package xsbti.compile;
 
 /**
-* Determines if an entry on a classpath contains a class.
-*/
-public interface DefinesClass
-{
- 	/**
-	* Returns true if the classpath entry contains the requested class.
-	*/
-	boolean apply(String className);
+ * Determines if a classpath entry contains a class.
+ *
+ * The corresponding classpath entry is not exposed by this interface. It's tied
+ * to it by a specific class that implements this interface.
+ */
+public interface DefinesClass {
+    /**
+     * Returns true if the classpath entry contains the requested class.
+     */
+    boolean apply(String className);
```

Comment:
Boolean isn't the best API for this, as confirming that something is defined will usually take a similar amount of time to just fetching it (similar to `Map.get` vs `Map.contains+Map.apply`).

Would prefer to see `Maybe`/`Option` here, or removal of this API.